libraries/AP_WindVane/AP_WindVane.cpp

```diff
@@ -206,35 +234,51 @@ void AP_WindVane::update()
         return;
     }
 
-    // check for calibration
-    calibrate();
+    // check if we should change backend
+    if (_init_direction_type != _direction_type || _init_speed_type != _speed_sensor_type) {
```

Comment:
we generally don't use deletes because it leads to memory fragmentation.  Instead we mark the type parameter as RebootRequired

Reply:
fixed